### PR TITLE
Use configuration providers instead of `envsubst` in Kafka configuration files

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

This PR changes how we handle various variables in the Kafka broker (or controller) configuration files. Instead of using environment variables and replacing them with `envsubst`, it moves to use configuration providers to have them loaded inside Kafka.

This helps us to:
* Minimize our dependency on bash scripts
* * Improves the test coverage as the bash scripts in the container images have no unit tests (they are of course run as part of system tests)
* Reduces our footprint in the container images (right now, this probably has little value, but the smaller footprint we have in the container image, the easier to support more different images)

The way this is implemented is that we initialize two different providers:
* The file provider (used only in broker nodes)
* The environment variable provider
* (We also respect and keep any providers configured by the user - although that might not be usual in Kafka brokers/controllers)

And we use these providers to read data from files or environment variables. The PKCS12 passwords and the OAuth secrets are passed through environment variables and rendered suing the environment variable provider. the rack ID and worker node address (used for node port listeners) are passed as a file and loaded through the file config provider. For this, the layout of the files generated by the init container was slightly modified,

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java [hunk 1]
```diff
@@ -821,6 +810,49 @@ private void configureCustomAuthorization(KafkaAuthorizationCustom authorization
...
+
+        if (userConfig != null
+                && !userConfig.getConfiguration().isEmpty()
+                && userConfig.getConfigOption("config.providers") != null) {
+            writer.println("# Configuration providers configured by the user and by Strimzi");
+            writer.println("config.providers=" + userConfig.getConfigOption("config.providers") + "," + strimziConfigProviders);
+            userConfig.removeConfigOption("config.providers");
+        } else {
+            //System.out.println("User config is: " + userConfig.getConfiguration());
+            //System.out.println("User config is: " + userConfig.getConfigOption("config.providers"));
```
Comment (im-konge) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java:838]: Is this some leftover from debug?

**Review by im-konge — APPROVED** [review]
LGTM, just one minor comment. Thanks :)

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java [hunk 2]
```diff
@@ -821,6 +810,49 @@ private void configureCustomAuthorization(KafkaAuthorizationCustom authorization
...
+            writer.println("config.providers=" + strimziConfigProviders);
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java:839]: Maybe adding a comment on the top like in the other "if" branch, something like:

```java
writer.println("# Configuration providers configured by Strimzi");
```

**Review by ppatierno — APPROVED** [review]
LGTM. Just a nit.

**Review by see-quick — APPROVED** [review]

**Pull request merged**